### ISSUE TITLE
parallel loading of d3 and c3 broken since rev ff3d590

### ISSUE THREAD

**Issue by Davidiusdadi** [c3js/c3#2060]
Commit https://github.com/c3js/c3/commit/ff3d5900f2ecc29fbdf2a0bb40707e23e8150ecd by @kt3k contains a (probably unintentional) breaking change:

Before https://github.com/c3js/c3/commit/ff3d5900f2ecc29fbdf2a0bb40707e23e8150ecd d3 had to be loaded **only before the first ChartInternal is created** (which happens when c3 is first used ).
After https://github.com/c3js/c3/commit/ff3d5900f2ecc29fbdf2a0bb40707e23e8150ecd d3 has to be loaded **before c3.js is first executed.**

The breaking change makes it impossible to load d3 and c3 in parallel which is odd as it breaks all code that relies on that behavior.

The following code is responsible:
```javascript
var d3 = window.d3 ? window.d3 : typeof require !== 'undefined' ? require("d3") : undefined;
```

Before there was no toplevel d3 object but the d3 referenced in ChartInternal 
```javascript
$$.d3 = window.d3 ? window.d3 : typeof require !== 'undefined' ? require("d3") : undefined;
```

In my opinion the old logic made much more sense as it allows for optimizing page loading times!

Just assigning the d3 object *first* in the ChartInternal and all should be fine.

**Comment by kt3k** [comment]
Thanks for pointing this! I agree with you that this should be fixed.

Actual load timing change of d3 seems to have happened at #2048.

**Comment by kt3k** [comment]
Published at v0.4.13.
